packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -238,6 +238,13 @@ class RenderEditable extends RenderBox {
   /// The default value of this property is false.
   bool ignorePointer;
 
+  /// Whether text is composed.
+  ///
+  /// Text is composed when user selects it for editing. The [TextSpan] will have
+  /// children with composing effect and leave text property to be null.
```

Comment:
I have tried to do the comparison directly in test but was advised to put it here. How would you move it to test?

Reply:
In the test you have

```dart
expect(renderEditable.isComposingText, false);
```

So that would become:

```dart
expect(renderEditable.text.text == null, false);
```

Or you could definitely move the whole separate method to the test file.

If someone else told you to move it here then it's probably fine though, I'm not sure what their reason was.  It just jumped out at me as something that didn't need to be exposed here.

Reply:
I think renderEditable.text.text is testing based implementation, if someone change the composing implementation, the will need to update the test.
If we expose isComposingText, the test will be purely testing the behavior. If someone change the implementation, they will need to update isComposingText method instead of touching test.

In this case, I am more toward exposing isComposingText, but I do get we should avoid exposing method for test only.

Reply:
That makes sense to me :+1: